### DISCUSSION TITLE
Storage Backend - Proxy Files

### DISCUSSION THREAD

**Discussion by shepherdjay** [netbox-community/netbox#13171] [Ideas]
### NetBox version

v3.5.4

### Feature type

Change to existing functionality

### Proposed functionality

Propose changing the way external files are provided back to user when STORAGE_BACKEND is in use. Currently when attempting to retrieve an image or file that was stored into a backend (for example GCP) the URL provided for GET request back to user is the storage_backend url. For example `https://storage.googleapis.com/mynetboxbucket/myimage.jpeg`

### Use case

Redirecting users directly to the storage backend adds complexity in permissions as not only do users require netbox permissions but they also require storage backend permissions to the bucket. 

Instead if Netbox can fetch and then return the stored file to the user only netbox requires access to the bucket.

**Comment by jsenecal** [comment]
Thank you for your interest in extending NetBox. Unfortunately, the information you have provided does not constitute an actionable feature request. Per our [contributing guide](https://github.com/netbox-community/netbox/blob/develop/CONTRIBUTING.md), a feature request must include a thorough description of the proposed functionality, including any database changes, new views or API endpoints, and so on. It must also include a detailed use case justifying its implementation. If you would like to elaborate on your proposal, please modify your post above. If sufficient detail is not added, this issue will be closed. 

Furthermore, I would like to add that Storage Backends rely on third party integrations/modules and as such nothing within netbox can be done to address this feature request as formulated.

**Comment by shepherdjay** [comment]
Will review as a github discussion per contributing guidlines

**Reply by jsenecal** [reply]
I migrated the issue to a discussion :)
